core/encoding/json/marshal.odin

```diff
@@ -263,36 +275,81 @@ marshal_to_writer :: proc(w: io.Writer, v: any, opt: ^Marshal_Options) -> (err:
 			map_cap := uintptr(runtime.map_cap(m^))
 			ks, vs, hs, _, _ := runtime.map_kvh_data_dynamic(m^, info.map_info)
 
-			i := 0
-			for bucket_index in 0..<map_cap {
-				runtime.map_hash_is_valid(hs[bucket_index]) or_continue
-
-				opt_write_iteration(w, opt, i) or_return
-				i += 1
+			if opt.sort_maps_by_key {
+				Entry :: struct {
+					key: string,
+					value: any,
+				}
 
-				key   := rawptr(runtime.map_cell_index_dynamic(ks, info.map_info.ks, bucket_index))
-				value := rawptr(runtime.map_cell_index_dynamic(vs, info.map_info.vs, bucket_index))
+				// If we are sorting the map by key, then we temp alloc an array
+				// and sort it, then output the result.
+				sorted := make([dynamic]Entry, 0, map_cap, context.temp_allocator)
```

Comment:
With this change a function that was previously allocating no memory would start allocating memory, causing some existing programs to leak memory. Maybe temp allocator guard should be introduced to free the temporary slice.

Reply:
Thanks. I introduced a temp alloc guard, its easy to get used to one's own setup with freeing the temp allocator per-frame.